### DISCUSSION TITLE
SubmitSelfServiceLoginFlowWithPasswordMethodBody

### DISCUSSION THREAD

**Discussion by alaghusun60** [ory/kratos#2691] [Q&A] [answered]
What is relevent api for this model in php. How we do this

**Comment by vinckr** [comment, marked as answer]
Hello @alaghusun60 
apologies for the late answer!
To integrate authentication into a PHP application using Ory, you need to follow these steps:

- Create a PHP app: You can create a new PHP project by running the following commands:

```
mkdir your-project  
cd your-project  
touch index.php
```

- Install dependencies: You need to install the Ory SDK and bramus/router for URL handling. You can do this by running the following commands:

```
composer require ory/client  
composer require bramus/router
```

- Create an Entry Page: You need to create an index.php script which creates an Ory client, registers a new route for your Dashboard and validates if the user is allowed to view the Dashboard. Here is a basic example:

```
<?php  
 require 'vendor/autoload.php';  
 require_once 'app.php';  
  
 error_reporting(E_ERROR | E_PARSE);  
  
 $proxyPort = getenv("PROXY_PORT");  
 if ($proxyPort == "") $proxyPort = "4000";  
  
 $app = new App;  
 $config = Ory\Client\Configuration::getDefaultConfiguration()->setHost(sprintf("http://localhost:%s/.ory", $proxyPort));  
 $app->ory = new Ory\Client\Api\FrontendApi(new GuzzleHttp\Client(), $config);  
  
 $router = new \Bramus\Router\Router();  
 $router->before('GET', '/', $app->validateSession());  
 $router->get('/', $app->printDashboard());  
 $router->run();  
?>
```

- Validate and Login: You need to create a handler which will check with your Ory project if the user has a valid session. If the session is not valid, the request is redirected to the Ory project for login. Here is an example:

```
<?php  
 class App {  
 private ?Ory\Client\Model\Session $session;  
 public ?Ory\Client\Api\FrontendApi $ory;  
  
 public function validateSession(){  
 $cookies = "";  
 foreach ($_COOKIE as $key=>$value) {  
 $cookies .= "$key=$value;";  
 }  
  
 try {  
 $session = $this->ory->toSession("", $cookies);  
 if (! $session["active"]) throw new Exception('Session expired');  
 } catch (Exception $e) {  
 error_log('Exception when calling toSession: '.$e->getMessage());  
 header("Location: /.ory/self-service/login/browser", true, 303);  
 die();  
 }  
 $this->session = $session;  
 }  
  
 public function printDashboard(){  
 echo '  
<html lang="en">  
<head>  
 <title>Ory Network secured Go web app</title>  
</head>  
<body>  
 <h1>Dashboard</h1>  
 <hr />  
 <h2>Your Session Data:</h2>  
 <pre><code>', json_encode($this->session, JSON_PRETTY_PRINT), '</code></pre>  
</body>  
</html>  
';  
 }  
 }  
?>
```

- Run your app: You can run your app by setting the environment variable to the exposed port and running the Ory proxy. Here is an example:

```
export ORY_SDK_URL=http://localhost:4433  
ory proxy http://localhost:3000
```

To access the PHP app through the ORY proxy, open http://localhost:4000/ in your browser. You are presented with Ory's Sign In page. Click on sign up and create your first user.
For more details, you can refer to the [Ory documentation](https://www.ory.sh/docs/getting-started/integrate)
